cli/lsp/sources.rs

```diff
@@ -248,28 +309,26 @@ impl Sources {
     None
   }
 
-  pub fn get_script_version(
+  fn get_script_version(
     &mut self,
     specifier: &ModuleSpecifier,
   ) -> Option<String> {
-    if let Some(path) = self.get_path(specifier) {
-      if let Ok(metadata) = fs::metadata(path) {
-        if let Ok(modified) = metadata.modified() {
-          return if let Ok(n) = modified.duration_since(SystemTime::UNIX_EPOCH)
-          {
-            Some(format!("{}", n.as_millis()))
-          } else {
-            Some("1".to_string())
-          };
+    let path = self.get_path(specifier)?;
+    if let Ok(metadata) = fs::metadata(path) {
```

Comment:
Just noting that this is doing synchronous file I/O while holding a lock. Probably not a big deal with local filesystems but network filesystems are a different story.

Aside: one more opportunity for simplifying the unboxing logic:
```suggestion
    let metadata = fs::metadata(path).ok()?;
```

Reply:
Thanks for the feedback.  I discussed it with @ry a while back and the biggest challenge is that doing it async will bubble up to the tsc ops, and all of the tsc ops are synchronous because of the way tsc works, and we know it breaks.  I think we can "hide" some of the asynchronous nature on the Rust side from tsc, but we didn't want to tackle that until we had a performance problem.

The first thing I think we would do is try to limit the number of version checks during a given operation, because right now, it could be checked several times in a row, though it is highly unlikely to have changed.